lib/matplotlib/axis.py

```diff
@@ -1743,7 +1745,9 @@ def _update_label_position(self, bboxes, bboxes2):
 
         else:
             if not len(bboxes2):
-                top = self.axes.bbox.ymax
+                top = (self.axes.spines['top'].get_transform().
```

Comment:
The docstring of the function only talks about tick labels, now the function uses spines too